File: PHP/Behavioral/Interpreter/Example/AddExpression.php
<?php
include_once 'Expression.php';
class AddExpression implements Expression
{
    private $leftExpression;
    private $rightExpression;

    public function __construct(Expression $left, Expression $right) {
        $this->leftExpression = $left;
        $this->rightExpression = $right;
    }

    public function interpret($context)
    {
        return $this->leftExpression->interpret($context) + $this->rightExpression->interpret($context);
    }
}

File: Csharp/Behavioral/Interpreter/Example3/Show/Show/AddExpression.cs
namespace Show
{
    public class AddExpression: IExpression
    {
        private IExpression leftExpression;
        private IExpression rightExpression;

        public AddExpression(IExpression left, IExpression right)
        {
            leftExpression = left;
            rightExpression = right;
        }
        
        public int Interpret(Context context)
        {
            return leftExpression.Interpret(context) + rightExpression.Interpret(context);
        }
    }
}